packages/bsb-core/bsb/config/refs.py

```diff
@@ -26,45 +48,78 @@ def up(self, here, to=None):
                 return None
         return here
 
+    def is_ref(self, value):
+        """
+        Check if the provided value corresponds to
+        the type of the reference
+
+        :param value: value to check
+        :rtype: bool
+        :return: True if the value has the type of the reference
+        """
+        return isinstance(value, self.type)
+
+    @property
+    def type(self):
+        """
+        Return the type of the reference
+        """
+        return None
+
+
+class FileReference(Reference):
+    def __call__(self, root, here):
+        return root.files
+
+    @property
+    def type(self):
+        from ..storage._files import FileDependencyNode
+
+        return FileDependencyNode
+
 
 class VoxelDatasetReference(Reference):
     def __call__(self, root, here):
-        return root.voxel_datasets
+        return {k: v for k, v in root.files.items() if isinstance(v, self.type)}
```

Comment:
This reference seems to return _all_ voxel datasets of a certain type? Also, filtering them this way means that the "remote" no longer is a configuration node but a dict, which I think is going to break a lot of assumptions. Can't this be better solved some other way?